Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx

```diff
@@ -0,0 +1,165 @@
+      }
+      offsetPrev += prev->mEntriesSlot[ch];
+      mEntriesSlot[ch] += prev->mEntriesSlot[ch];
+    }
+  }
+  mFT0CollectedCalibInfoSlot.swap(tmpVector);
+  LOG(DEBUG) << "After merging the size is " << mFT0CollectedCalibInfoSlot.size();
+  return;
+}
+//_____________________________________________
+void FT0CalibInfoSlot::print() const
+{
+  // to print number of entries in the tree and the channel with the max number of entries
+
+  LOG(INFO) << "Total number of entries " << mFT0CollectedCalibInfoSlot.size();
+  auto maxElementIndex = std::max_element(mEntriesSlot.begin(), mEntriesSlot.end());
+  auto channelIndex = std::distance(mEntriesSlot.begin(), maxElementIndex);
+  LOG(INFO) << "The maximum number of entries per channel in the current mFT0CollectedCalibInfo is " << *maxElementIndex << " for channel " << channelIndex;
+  return;
+}
+
+//_____________________________________________
+void FT0CalibInfoSlot::printEntries() const
+{
+  // to print number of entries in the tree and per channel
+
+  LOG(DEBUG) << "Total number of entries " << mFT0CollectedCalibInfoSlot.size();
+  for (int i = 0; i < mEntriesSlot.size(); ++i) {
+    if (mEntriesSlot[i] != 0) {
+      LOG(INFO) << "channel " << i << " has " << mEntriesSlot[i] << " entries";
+    }
+  }
+  return;
+}
+
+//===================================================================
+
+//_____________________________________________
+void FT0CalibCollector::initOutput()
+{
+  // emptying the vectors
+
+  mFT0CollectedCalibInfo.clear();
+  for (int ch = 0; ch < NCHANNELS; ch++) {
+    mEntries[ch] = 0;
+  }
+
+  return;
+}
+
+//_____________________________________________
+bool FT0CalibCollector::hasEnoughData(const Slot& slot) const
+{
+
+  // We define that we have enough data if the tree is big enough.
+  // each FT0CalibrationInfoObject is composed of one int, two floats --> 12 bytes
+  // E.g. supposing that we have 256 entries per channel (which is an upper limit ) --> ~523 MB
+  // we can check if we have at least 1 GB of data --> 500*o2::ft0::Geometry::NCHANNELS entries in the vector
+  // (see header file for the fact that mMaxNumOfHits = 500)
+  // The case in which mScaleMaxNumOfHits = false allows for a fast check
+
+  if (mTest) {
+    return true;
+  }
+  const o2::ft0::FT0CalibInfoSlot* c = slot.getContainer();
+  LOG(INFO) << "we have " << c->getCollectedCalibInfoSlot().size() << " entries";
+  int maxNumberOfHits = mAbsMaxNumOfHits ? mMaxNumOfHits : mMaxNumOfHits * NCHANNELS;
+  if (mTFsendingPolicy || c->getCollectedCalibInfoSlot().size() > maxNumberOfHits) {
+    return true;
+  }
+  return false;
+}
+
+//_____________________________________________
+void FT0CalibCollector::finalizeSlot(Slot& slot)
+{
+  // here we fill the tree with the remaining stuff that was not filled before
+
+  o2::ft0::FT0CalibInfoSlot* c = slot.getContainer();
+  mFT0CollectedCalibInfo = c->getCollectedCalibInfoSlot();
```

Comment:
How your slot is defined? Do you have many or just 1? Note that in principle, with multiple slots the TimeSlotCalibration may finalize multiple slots while processing a single TF (e.g. if there was pause, the newly arrived TF may trigger finalization of multiple slots). In this case you will store to the writer only the last slot.